Core/Graphics/Include/Babylon/Graphics.h

```diff
@@ -0,0 +1,42 @@
+#pragma once
+
+#include <Babylon/JsRuntime.h>
+
+#include <memory>
+
+namespace Babylon
+{
+    class Graphics
+    {
+    public:
+        class Impl;
+
+        ~Graphics();
+
+        template<typename NativeWindowT>
+        static std::unique_ptr<Graphics> InitializeFromWindow(NativeWindowT window, size_t width, size_t height);
```

Comment:
If I call ```InitializeFromWindow<HWND>``` in my code, I'll get a linker error I guess because the template specialization is only done for void* but it will compile just fine.

Reply:
Correct. This is a technique we created to allow platform-specific types to be eliminated from platform-agnostic header files while still having the build-time type safety (not compile-time, but link-time) that would be lost by just casting things to `void*`. I think we first used this pattern somewhere in XR, but it's currently most prominently used for [`Napi::Attach(...)`](https://github.com/BabylonJS/BabylonNative/blob/master/Dependencies/napi/napi-direct/include/napi/env.h#L7). Intellisense can't figure much out from this, but it remains safe without the overhead of either a parameter struct or multiple headers with a config-time swapping mechanism in the build system.

Reply:
To clarify, specializing this for `void*` is an idiotic thing to do that completely discards the aforementioned benefits of this pattern, and that's not what it's supposed to do long-term. 😄 I just haven't gotten around to doing the actual correct specializations yet; and until I do, the `void*` specialization emulates the prior behavior, which was unsafe but still worked.